src/ImageMetadata.jl

```diff
@@ -66,87 +77,100 @@ AxisArrays.HasAxes(A::ImageMetaAxis) = AxisArrays.HasAxes{true}()
         @inline function Base.setindex!(img::$AType, val, i::Int)
-            @boundscheck checkbounds(img.data, i)
-            @inbounds img.data[i] = val
+            @boundscheck checkbounds(data(img), i)
+            @inbounds data(img)[i] = val
             val
         end
         @inline function Base.setindex!(img::$AType{T,N}, val, I::Vararg{Int,N}) where {T,N}
-            @boundscheck checkbounds(img.data, I...)
-            @inbounds img.data[I...] = val
+            @boundscheck checkbounds(data(img), I...)
+            @inbounds data(img)[I...] = val
             val
         end
     end
 end
 
 @inline function Base.getindex(img::ImageMetaAxis, ax::Axis, I...)
-    result = img.data[ax, I...]
+    result = data(img)[ax, I...]
     maybe_wrap(img, result)
 end
 @inline function Base.getindex(img::ImageMetaAxis, i::Union{Integer,AbstractVector,Colon}, I...)
-    result = img.data[i, I...]
+    result = data(img)[i, I...]
     maybe_wrap(img, result)
 end
 maybe_wrap(img::ImageMeta{T}, result::T) where T = result
 maybe_wrap(img::ImageMeta{T}, result::AbstractArray{T}) where T = copyproperties(img, result)
 
 
 @inline function Base.setindex!(img::ImageMetaAxis, val, ax::Axis, I...)
-    setindex!(img.data, val, ax, I...)
+    setindex!(data(img), val, ax, I...)
 end
 @inline function Base.setindex!(img::ImageMetaAxis, val, i::Union{Integer,AbstractVector,Colon}, I...)
-    setindex!(img.data, val, i, I...)
+    setindex!(data(img), val, i, I...)
 end
 
-Base.view(img::ImageMeta, ax::Axis, I...) = shareproperties(img, view(img.data, ax, I...))
-Base.view(img::ImageMeta{T,N}, I::Vararg{ViewIndex,N}) where {T,N} = shareproperties(img, view(img.data, I...))
-Base.view(img::ImageMeta, i::ViewIndex) = shareproperties(img, view(img.data, i))
-Base.view(img::ImageMeta, I::Vararg{ViewIndex,N}) where {N} = shareproperties(img, view(img.data, I...))
+Base.view(img::ImageMeta, ax::Axis, I...) = shareproperties(img, view(data(img), ax, I...))
+Base.view(img::ImageMeta{T,N}, I::Vararg{ViewIndex,N}) where {T,N} = shareproperties(img, view(data(img), I...))
+Base.view(img::ImageMeta, i::ViewIndex) = shareproperties(img, view(data(img), i))
+Base.view(img::ImageMeta, I::Vararg{ViewIndex,N}) where {N} = shareproperties(img, view(data(img), I...))
+
+function Base.getproperty(img::ImageMeta, propname::Symbol)
+    # TODO remove these once deprecations are done
+    if propname === :data
+        return getproperty_data(img)
+    elseif propname === :properties
+        return getproperty_properties(img)
+    else
+        return properties(img)[propname]
+    end
+end
 
-Base.getindex(img::ImageMeta, propname::AbstractString) = img.properties[propname]
+function Base.setproperty!(img::ImageMeta, propname::Symbol, X)
+    return setindex!(properties(img), X, propname )
+end
 
-Base.setindex!(img::ImageMeta, X, propname::AbstractString) = setindex!(img.properties, X, propname)
+Base.propertynames(img::ImageMeta) = (keys(properties(img))...,)
```

Comment:
Is it safe deleting this method?

Reply:
I don't see why it wouldn't be.  In a way it's actually safer to not use `setindex!` for the properties. On the off chance that the underlying array uses a `String` as a key in it's index `setindex!` would make it impossible to index into the underlying structure. It would just keep redirecting to the `properties` field.

Reply:
I don't quite see how `setindex!` would be used here, and it looks like a non-trival change, just asked this to make sure it's not changed incidentally. I have to admit that I haven't got a chance to use this package 😂 

I trust your explanations here.